src/segmentation_display_state/property_map.ts

```diff
@@ -877,16 +881,50 @@ export function executeSegmentQuery(
     }
     indices = indices.subarray(0, outIndex);
   };
+  const filterByTagDescriptions = (regexp: RegExp) => {
+    const tagDescriptions = db!.tags!.tagDescriptions!;
+    const tags = db!.tags!.tags!;
+    const matchingTagDescriptions = tagDescriptions
```

Comment:
It would probably be more efficient to just iterate over the tags with a for loop --- then we wouldn't need to create all of these temporary arrays.

Reply:
I think i've addressed this.. it's failing tests for unrelated reasons.